chips/lowrisc/src/i2c.rs

```diff
@@ -270,15 +270,16 @@ impl<'a> I2c<'_> {
                     break;
                 }
                 // Send the data
-                regs.fdata.write(FDATA::FBYTE.val(buf[i as usize] as u32));
+                regs.fdata
+                    .write(FDATA::FBYTE.val(*buf.get(i).unwrap_or(&0) as u32));
                 data_pushed = i;
             }
 
             // Check if we can send the last byte
             if regs.status.read(STATUS::FMTFULL) == 0 && data_pushed == (len - 1) {
                 // Send the last byte with the stop signal
                 regs.fdata
-                    .write(FDATA::FBYTE.val(buf[len as usize] as u32) + FDATA::STOP::SET);
+                    .write(FDATA::FBYTE.val(*buf.get(len).unwrap_or(&0) as u32) + FDATA::STOP::SET);
```

Comment:
Not sure if it's better to panic in a case like this or to 'silently fail' and write 0?

Reply:
If you just care about binary size, usually adding a
```rust
if len >= buf.len() { return; } 
```
at the top and then keeping the normal array access should be sufficient for LLVM to optimize out the later panicking bounds check. 

When I made a similar series of passes to a driver for code size, I preferred to add Result<(), ErrorCode> return values to these functions and then just panic in the event of any error at the highest level possible, which meant that attempts at OOB accesses would still generate panics, but would not force a size cost of 12 panics in the binary and instead just have the size cost of 1.

Reply:
> If you just care about binary size, usually adding a
> 
> ```rust
> if len >= buf.len() { return; } 
> ```
> 
> at the top and then keeping the normal array access should be sufficient for LLVM to optimize out the later panicking bounds check.
> 
Ah cool!
> When I made a similar series of passes to a driver for code size, I preferred to add Result<(), ErrorCode> return values to these functions and then just panic in the event of any error at the highest level possible, which meant that attempts at OOB accesses would still generate panics, but would not force a size cost of 12 panics in the binary and instead just have the size cost of 1.

Sounds like a better idea, I'll try to implement these where possible.